src/Sylius/Behat/Context/Ui/Admin/ManagingPromotionsContext.php

```diff
@@ -617,6 +620,30 @@ public function thePromotionsShouldHavePriority(PromotionInterface $promotion, i
         Assert::same($this->updatePage->getPriority(), $priority);
     }
 
+    /**
+     * @When /^I want to manage (this promotion)'s coupons$/
+     */
+    public function iWantToManageThisPromotionSCoupons(): void
+    {
+        Assert::true($this->updatePage->isCouponManagementAvailable());
+    }
+
+    /**
+     * @Then I should not be able to access coupons management page
+     */
+    public function iShouldNotBeAbleToAccessCouponsManagementPage(): void
+    {
+        Assert::false($this->updatePage->isCouponManagementAvailable());
+    }
+
+    /**
+     * @Then /^I should be on (this promotion)'s coupons management page$/
+     */
+    public function iShouldBeOnThisPromotionSCouponsManagementPage(PromotionInterface $promotion): void
+    {
+        $this->indexPage->open(['id' => $promotion->getId()]);
```

Comment:
And in **this** step we should check if we're on the proper page (by calling `indexPage->isOpen(...`, for example